plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -73,6 +73,11 @@ type predicateMetadata struct {
 	podRequest                *schedulercache.Resource
 	podPorts                  map[int]bool
 	matchingAntiAffinityTerms []matchingPodAntiAffinityTerm
+
+	// CheckServiceAffinity: pods cache, otherwise this is a large query, will slow things down
+	// when there are 100s of services/pods.
+	lock        sync.Mutex
```

Comment:
hmmm, i guess if we make it a generic PreCompute maybe other predicates could reuse, ill think about this and see whats possible.  i was assuming lazy+decoupled would be the simplest way forward.  

My original thinking was that we may want to mutate the servicePods to nil at some point to force recompute, i.e. because there might be more than one way to compute affinity, and we might want to redo the computation in a cluster thats running.... but I guess `findNodesThatFit` will be recalled anytime pods die off, so the affinity will be naturally recalculated by the scheduler on an as needed basis.